client/src/components/ContentSelector/index.js

```diff
@@ -0,0 +1,39 @@
+import ToggleButton from "../ToggleButton"
+
+function ContentSelector(props) {
+    const {selected, setSelected} = props
+
+    const toggleShark = () => {
+        // Make sure that at least one option is always selected
+        if (selected.sharks && !selected.cats) {
+            setSelected({sharks: false, cats: true})
+        } else {
+            setSelected({...selected, sharks: !selected.sharks})
+        }
+    }
+
+    const toggleCat = () => {
+        if (selected.cats && !selected.sharks) {
+            setSelected({sharks: true, cats: false})
+        } else {
+            setSelected({...selected, cats: !selected.cats})
+        }
+    }
```

Comment:
How would you refactor this into a single function?

Reply:
One feature I thought would be nice is to have it so trying to unselect a category when doing so would result in no categories being selected would instead switch the selection to the other category (as can be seen in the logic found in the quoted lines of code). The way I'd refactor this would depend on what we'd want to do about this feature.

Retaining this feature, I would refactor as follows:
```js
const canToggle = (key) => {
    // Check that this key is not (going to be toggled off and the only one selected)
    const numTrues = Object.values(selected).reduce((count, current) => {
        return count + (current ? 1 : 0)
    })

    return !selected[key] || numTrues > 1
}

const toggle = (key) => {
    // Ensure at least one option is always selected
    if (canToggle(key)) {
        setSelected({...selected, [key]: !selected[key]})
    } else {
        // If there are more than two options, this behaviour will likely
        //      feel arbitrary and confusing to the user.
        const otherKey = Object.keys(selected).filter(k => k !== key)[0]
        setSelected({[key]: false, [otherKey]: true})
    }
}
```

Otherwise, I would refactor it like this:
```js
const canToggle = (key) => {
    // Check that this key is not (going to be toggled off and the only one selected)
    const numTrues = Object.values(selected).reduce((count, current) => {
        return count + (current ? 1 : 0)
    })

    return !selected[key] || numTrues > 1
}

const toggle = (key) => {
    // Ensure at least one option is always selected
    if (canToggle(key)) {
        setSelected({...selected, [key]: !selected[key]})
    }
}
```